### ISSUE TITLE
Signatures - Clarify usage of plus character when encoding query parameters

### ISSUE THREAD

**Issue by robelcik** [httpwg/http-extensions#2641]
Hello,

In the section **2.2.8. Query Parameters** we can read that:
> Encode the nameString or valueString using the "percent-encode after encoding" process defined by the "application/x-www-form-urlencoded serializing" section of [[HTMLURL](https://url.spec.whatwg.org/#application/x-www-form-urlencoded)], which results in an [[ASCII](https://www.rfc-editor.org/rfc/rfc20)] string.

As I understand the referred specification, spaces should be replaced by plus character, which is indicated by the last word below, which is "true":
(note, that it's different than encoding described in RFC 3986)
> Let value be the result of running [percent-encode after encoding](https://url.spec.whatwg.org/#string-percent-encode-after-encoding) with encoding, tuple’s value, the [application/x-www-form-urlencoded percent-encode set](https://url.spec.whatwg.org/#application-x-www-form-urlencoded-percent-encode-set), and true.

However, the example in 2.2.8. shows that "+" characters are replaced by "%20".

Could you please, clarify it?

Also, it would be nice, I think, to add more examples. For example "raw" GET requests containing sequences like "-", "%2D", "%48".

Thank you in advance for reply.

Regards,
Robert

**Comment by robelcik** [comment]
Hello!

Thank you for explanation. Now I understand your intentions, 😃 

However, I'd like to nitpick a bit. I asked the initial question because I was confused. In section 2.2.8., there is a reference to "[application/x-www-form-urlencoded serializing](https://url.spec.whatwg.org/#urlencoded-serializing)". That section says, as far as I understand, that _spaceAsPlus_ should be _true_.
![image](https://github.com/httpwg/http-extensions/assets/12409950/0432ab4d-5094-445b-9a69-a5f6e17c3ae0)

When looking at section "[Percent-encoded bytes](https://url.spec.whatwg.org/#percent-encoded-bytes)", one can see that default value of _spaceAsPlus_ is _false_ which, I assume, is what you refer to speaking of "canonical encoding".
![image](https://github.com/httpwg/http-extensions/assets/12409950/85b3c362-e5cd-4efb-860e-75c2eeb80a7b)

I was thinking that someone could get confused as I was, 🤓 

Regards,
Robert

**Comment by jricher** [comment]
Yes, the intent in the text is to refer to the percent encode after encoding with the space as plus set to false, its default value. I can understand the confusion here, so I'll see if we can add any clarifying text to the section during final RFC editor review, whenever that happens.
